python_modules/dagster/dagster/_core/definitions/resource_annotation.py

```diff
@@ -20,12 +21,28 @@ def get_resource_args(fn) -> Sequence[Parameter]:
 RESOURCE_PARAM_METADATA = "resource_param"
 
 
+class UsableAsResourceParam(ABC):
```

Comment:
I wonder if the name should convey that this allows the class to be usable unannotated as a resource param. Maybe something like `InferAsResourceParam`, `TreatAsResourceParam`, `TreatParamAsResource`?

In the default case, we don't assume an input class annotation `MyClass` is a resource because it might be the output of e.g. an asset. Still, `MyClass` is already "usable as a resource param" if specified as `ResourceParam[MyClass]`.

Reply:
Agree with this feedback. I like `TreatAsResourceParam`. I will rename and land.